collectors/python.d.plugin/tomcat/tomcat.chart.py

```diff
@@ -104,11 +127,18 @@ def _get_data(self):
         data = None
         raw_data = self._get_raw_data()
         if raw_data:
-            try:
-                xml = ET.fromstring(raw_data)
-            except ET.ParseError:
-                self.debug('%s is not a vaild XML page. Please add "?XML=true" to tomcat status page.' % self.url)
+            if fix_tomcat_single_quote and single_quote_regex.search(raw_data):
+                self.warning('Tomcat status page is returning invalid single quote XML, please consider upgrading '
+                             'your Tomcat installation. See https://bz.apache.org/bugzilla/show_bug.cgi?id=61603')
```

Comment:
i think we need to add `check` method to `Service` and select appropiate parser there.
`check` method executed only once.

you need 

```python
def check(self):
    self._manager = self._build_manager()  # this is needed

    raw_data = self._get_raw_data()
    if not raw_data:
        return None

     <parser select>
     <assigne parser to self.parse>

     <if selected return true, otherwise false>
```


and i think we don'nt need parser as Classes, simpe funcs will do

Reply:
Something like that ? (cf. next commit)

I'm not sure I understood the `<if selected return true, otherwise false>` part. I always return True as long as the `raw_data` check is ok ; otherwise it does not work.